src/integration/integration_test.go

```diff
@@ -0,0 +1,123 @@
+		Expect(err).NotTo(HaveOccurred())
+
+		tlsCert := otelConfigVars.TLSCert()
+		testOtelReceiver = grpc.NewServer(grpc.Creds(credentials.NewServerTLSFromCert(&tlsCert)))
+
+		fakeMetricsServiceServer = NewFakeMetricsServiceServer()
+		colmetricspb.RegisterMetricsServiceServer(testOtelReceiver, fakeMetricsServiceServer)
+		go testOtelReceiver.Serve(lis)
+
+		conn, err := grpc.NewClient(fmt.Sprintf("127.0.0.1:%d", otelConfigVars.IngressOTLPPort),
+			grpc.WithTransportCredentials(otelConfigVars.CaAsGrpcTransportCredentials()),
+		)
+		Expect(err).NotTo(HaveOccurred())
+		msc = colmetricspb.NewMetricsServiceClient(conn)
+	})
+
+	JustBeforeEach(func() {
+		var err error
+		configyml, err := os.ReadFile(fmt.Sprintf("testdata/%s", otelConfigPath))
+		Expect(err).NotTo(HaveOccurred())
+		t, err := template.New("test").Parse(string(configyml))
+		Expect(err).NotTo(HaveOccurred())
+
+		buf := new(bytes.Buffer)
+		err = t.Execute(buf, otelConfigVars)
+		Expect(err).NotTo(HaveOccurred())
+		fmt.Println(buf.String())
+		os.WriteFile(TempFilePath("config.yml"), buf.Bytes(), 0660)
+
+		ctx = context.TODO()
+		cmd := exec.CommandContext(ctx, componentPaths.Collector, fmt.Sprintf("--config=file:%s", TempFilePath("config.yml")))
+		otelCollectorSession, err = gexec.Start(cmd, GinkgoWriter, GinkgoWriter)
+		Expect(err).NotTo(HaveOccurred())
+		Eventually(ctx, otelCollectorSession.Err).Should(gbytes.Say(`Everything is ready. Begin running and processing data.`))
+	})
+
+	JustAfterEach(func() {
+		otelCollectorSession.Kill()
+		testOtelReceiver.Stop()
+	})
+
+	Describe("metrics", func() {
+		BeforeEach(func() {
+			otelConfigPath = "simple.yml"
+		})
+
+		It("can forward an otlp metric", func() {
+			rm := NewSimpleResourceMetrics("goes_up")
+			_, err := msc.Export(ctx, &colmetricspb.ExportMetricsServiceRequest{
+				ResourceMetrics: []*metricspb.ResourceMetrics{&rm},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			var emsr *colmetricspb.ExportMetricsServiceRequest
+			Eventually(ctx, fakeMetricsServiceServer.ExportMetricsServiceRequests, 5).Should(Receive(&emsr))
+			Expect(emsr.GetResourceMetrics()[0].GetScopeMetrics()[0].GetMetrics()[0].Name).To(Equal("goes_up"))
+		})
+
+		Context("when a prometheus exporter is configured", func() {
+			BeforeEach(func() {
+				otelConfigPath = "prometheus_exporter.yml"
+			})
+
+			It("serves the metric over the promql endpoint", func() {
+				rm := NewSimpleResourceMetrics("goes_down")
+				_, err := msc.Export(ctx, &colmetricspb.ExportMetricsServiceRequest{
+					ResourceMetrics: []*metricspb.ResourceMetrics{&rm},
+				})
+				Expect(err).NotTo(HaveOccurred())
+
+				client := &http.Client{
+					Transport: &http.Transport{
+						TLSClientConfig: otelConfigVars.CaAsTLSConfig(),
+					},
+				}
+
+				Eventually(func() []byte {
+					response, err := client.Get(fmt.Sprintf("https://127.0.0.1:%d/metrics", otelConfigVars.Port))
+					Expect(err).NotTo(HaveOccurred())
```

Comment:
There is a way to use Gomega expectations inside an eventually. This looks like it might behave unexpectedly. The docs say that to pass a local Gomega, e.g.
```Golang
Eventually(func(g Gomega) []byte {
  ...
  g.Expect(err).NotTo(HaveOccurred())
```